include/RAJA/policy/sycl/launch.hpp

```diff
@@ -176,7 +174,7 @@ struct LaunchExecute<RAJA::sycl_launch_t<async, 0>> {
             ctx.itm = &itm;
 
             //Point to shared memory
-            ctx.shared_mem_ptr = s_vec.get_pointer().get();
+            ctx.shared_mem_ptr = s_vec.get_multi_ptr<::sycl::access::decorated::yes>().get();
```

Comment:
Hi @homerdin, I was trying to eliminate SYCL warnings from RAJA (and CAMP https://github.com/LLNL/camp/pull/147), and had to update some of our SYCL functionality to use the 2020 API. Most of the changes were straightforward, except this new `get_multi_ptr`. Specifically, I couldn't find documentation on what `access::decorated` type to pass in; `yes` seems to work, but it would be good to know what exactly this is doing and whether this actually gets us the shared memory we are expecting from this call. Please let me know if this is ok, and feel free to change this to something better.

Reply:
From the [sycl 2020 specs](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#command): 

> The multi_ptr class exposes 3 flavors of the same interface. If the value of access::decorated is access::decorated::no, the interface exposes pointers and references type that are not decorated by an address space. If the value of access::decorated is access::decorated::yes, the interface exposes pointers and references type that are decorated by an address space. The decoration is implementation dependent and relies on device compiler extensions. The decorated type may be distinct from the non-decorated one. For interoperability with the [SYCL backend](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#backend), users should rely on types exposed by the decorated version. If the value of access::decorated is access::decorated::legacy, the 1.2.1 interface is exposed. This interface is deprecated.

Reply:
Thanks @long58, that's what I was seeing as well, and was hoping for a better explanation of `yes`, `no`, and `legacy`.

Reply:
FYI, I just tried this branch and the corresponding camp branch in a RAJAPerf build on corona and it built and ran correctly. I'll have a RAJAPerf branch ready to merge after the RAJA and camp PRs are merged.

"Three flavors"....who writes their documentation?!? Although the code changes to taste a bit like "yes" 😄

Reply:
@long58 thanks for citing the SYCL spec reference.

Reply:
@rchen20 I was told that they are adding the corresponding __local, __private, __global and __constant to pointer/reference declarations.  `yes` should be correct.

Reply:
> @rchen20 I was told that they are adding the corresponding __local, __private, __global and __constant to pointer/reference declarations. `yes` should be correct.

Thanks! To be clear, do you mean that `get_multi_ptr<::sycl::access::decorated::yes>` returns a pointer or reference decorated with one of the above qualifiers (depending on the base pointer/ref type)?